pyomo/gdp/plugins/bound_pretransformation.py

```diff
@@ -0,0 +1,337 @@
+
+    def _apply_to(self, instance, **kwds):
+        if not instance.ctype in (Block, Disjunct):
+            raise GDP_Error(
+                "Transformation called on %s of type %s. 'instance'"
+                " must be a ConcreteModel, Block, or Disjunct (in "
+                "the case of nested disjunctions)." % (instance.name, instance.ctype)
+            )
+
+        self._config = self.CONFIG(kwds.pop('options', {}))
+        self._config.set_value(kwds)
+
+        targets = self._config.targets
+        if targets is None:
+            targets = (instance,)
+
+        transformation_blocks = {}
+        bound_dict = ComponentMap()
+        self._update_bounds_from_constraints(instance, bound_dict, None, is_root=True)
+        # [ESJ 05/04/23]: In the future, I should think about getting my little
+        # trees from this tree, or asking for leaves rooted somewhere specific
+        # or something. Because this transformation currently does the work of
+        # getting the GDP tree twice...
+        whole_tree = get_gdp_tree(targets, instance)
+        for t in whole_tree.topological_sort():
+            if t.ctype is Disjunction and whole_tree.in_degree(t) == 0:
+                self._transform_disjunction(
+                    t, instance, bound_dict, transformation_blocks
+                )
+
+    def _transform_disjunction(
+        self, disjunction, instance, bound_dict, transformation_blocks
+    ):
+        # We go from root to leaves so that whenever we hit a variable, we can
+        # ask if the bounds we're seeing on its parent Disjunct (or if we're at
+        # the root, in the global scope) are looser or tighter than the bounds
+        # on it, and we pass the tightest ones down. For sane models, the bounds
+        # will tighten as we go down the tree, but that's of course not
+        # guaranteed since not all models are sane...
+        disjunctions_to_transform = set()
+        gdp_forest = get_gdp_tree((disjunction,), instance)
+        for d in gdp_forest.topological_sort():
+            if d.ctype is Disjunct:
+                self._update_bounds_from_constraints(d, bound_dict, gdp_forest)
+        self._create_transformation_constraints(
+            disjunction, bound_dict, gdp_forest, transformation_blocks
+        )
+
+    def _get_bound_dict_for_var(self, bound_dict, v):
+        v_bounds = bound_dict.get(v)
+        if v_bounds is None:
+            v_bounds = bound_dict[v] = {None: (v.lb, v.ub), 'to_deactivate': set()}
+        return v_bounds
+
+    def _update_bounds_from_constraints(
+        self, disjunct, bound_dict, gdp_forest, is_root=False
+    ):
+        bound_dict_key = None if is_root else disjunct
+        for constraint in disjunct.component_data_objects(
+            Constraint,
+            active=True,
+            descend_into=Block,
+            sort=SortComponents.deterministic,
+        ):
+            if hasattr(constraint.body, 'ctype') and constraint.body.ctype is Var:
```

Comment:
Why do you need this special case?  `identify_variables()` will still return a single Var, so the next branch should still work.

Reply:
I handled it separately because if this is true I know that I don't need to bother with using `generate_standard_repn` (or doing any algebra at all). But I'm fine with not doing it separately--just didn't know if it would be enough to effect performance?